sycl/test/tracing/enhanced_tracing_test.cpp

```diff
@@ -0,0 +1,49 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: env SYCL_PI_TRACE=-1 %CPU_RUN_PLACEHOLDER %t.out %CPU_CHECK_PLACEHOLDER
+// RUN: env SYCL_PI_TRACE=-1 %GPU_RUN_PLACEHOLDER %t.out %GPU_CHECK_PLACEHOLDER
+// RUN: env SYCL_PI_TRACE=-1 %ACC_RUN_PLACEHOLDER %t.out %ACC_CHECK_PLACEHOLDER
+
+// Test to check enhanced PI tracing
+
+// CHECK: ---> piMemBufferCreate(
+// CHECK-NEXT: <unknown> : {{0[xX]?[0-9a-fA-F]*}}
+// CHECK-NEXT: <unknown> : 1
+// CHECK-NEXT: <unknown> : 40
+// CHECK-NEXT: <unknown> : 0
+// CHECK-NEXT: <unknown> : {{0[xX]?[0-9a-fA-F]*}}
+// CHECK-NEXT: ) ---> 	pi_result : PI_SUCCESS
+// CHECK-NEXT: [out]void * : {{0+}}
+// CHECK-NEXT: [out]pi_mem * : {{0[xX]?[0-9a-fA-F]*}}[ {{0[xX]?[0-9a-fA-F]*}}
+//
+// CHECK: ---> piEnqueueKernelLaunch(
+// CHECK-NEXT: <unknown> : {{0[xX]?[0-9a-fA-F]*}}
+// CHECK-NEXT: <unknown> : {{0[xX]?[0-9a-fA-F]*}}
+// CHECK-NEXT: <unknown> : 1
+// CHECK-NEXT: <unknown> : {{0[xX]?[0-9a-fA-F]*}}
+// CHECK-NEXT: <unknown> : {{0[xX]?[0-9a-fA-F]*}}
+// CHECK-NEXT: <unknown> : 0
+// CHECK-NEXT: <unknown> : 0
+// CHECK-NEXT: pi_event * : {{0+}}[ nullptr ]
```

Comment:
could you cover all introduced data type variants: pi_image_offset, pi_buff_rect_offset , etc?

Reply:
Added tests to cover all type variants.